unittests/APITests/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+set (TESTS
+  ReentrantMutex)
+
+list(APPEND LIBS FEXCore)
+
+foreach(API_TEST ${TESTS})
+  add_executable(${API_TEST} ${API_TEST}.cpp)
+  target_link_libraries(${API_TEST} PRIVATE ${LIBS} Catch2::Catch2WithMain)
+
+  catch_discover_tests(${API_TEST}
+    TEST_SUFFIX ".APITest")
```

Comment:
Wouldn't it make more sense to make the "toplevel" category a prefix of the test name rather than a suffix? Not sure if we have a consistent scheme for existing tests, but would be great to settle for a unified scheme for Catch2 tests going forward.

Reply:
Also, how about using `".${API_TEST}.APITest"` instead? That way, we don't need to repeat "Reentrant" for each of the `TEST_CASE`s below.

Reply:
We have consistent scheme for the unit tests as they are all suffix based.
IR tests: ".ir"
ASM tests: ".asm"
32bit ASM tests: ".asm", but also scans by 32Bit in the filename. Should be changed.
gvisor tests: ".gvisor"
Posix tests: ".posix"
32-bit gcc tests: ".gcc-target-32"
64-bit gcc tests: ".gcc-target-64"

Reply:
hmm suffix-syntax for test categories looks very alien to me, but maybe I just need to get used to it :)

Good to know it's mostly consistent already. I'll adapt the thunking PR to use the same scheme then.

Reply:
Did you have opinions on using `.${API_TEST}.APITest` for test names instead?

Currently, the test list doesn't contain references to the tested class:
```
    Start 37: SimpleWait.APITest
    Start 38: SignaledWait.APITest
    Start 39: SignaledWaitLongJumpNoSignal.APITest
    Start 40: SignaledWaitLongJumpSignal.APITest
1/4 Test #38: SignaledWait.APITest ...................   Passed    0.89 sec
2/4 Test #40: SignaledWaitLongJumpSignal.APITest .....   Passed    0.89 sec
3/4 Test #37: SimpleWait.APITest .....................   Passed    1.88 sec
4/4 Test #39: SignaledWaitLongJumpNoSignal.APITest ...   Passed    2.89 sec
```

With the suggested change, e.g. the first test name should become `SignaledWait.ReentrantMutex.APITest` instead, so it's more apparent what the actual entity under test is.

Reply:
Changed over to subcategory naming.